src/res_parsetree_viewer.mli

```diff
@@ -17,6 +17,13 @@ val functorType :
 val processUncurriedAttribute :
   Parsetree.attributes -> bool * Parsetree.attributes
 
+(* determines whether a function is async and/or uncurried based on the given attributes *)
+val processFunctionAttributes :
+  Parsetree.attributes ->
+  bool (* async *) * bool (* uncurried *) * Parsetree.attributes
```

Comment:
Could maybe return a record instead?

Reply:
Implemented in https://github.com/rescript-lang/syntax/pull/600/commits/877145c4c9c7207cdbd831098676bcf9cda40c5d